ipa-core/src/protocol/basics/reveal.rs

```diff
@@ -73,6 +78,52 @@ pub trait Reveal<C: Context> {
         C: 'fut;
 }
 
+impl<C, S> Reveal<C> for [S]
+where
+    C: Context,
+    S: Reveal<C> + Send + Sync + 'static,
+{
+    type Output = Vec<<S as Reveal<C>>::Output>;
+
+    fn generic_reveal<'fut>(
+        &'fut self,
+        ctx: C,
+        record_id: RecordId,
+        excluded: Option<Role>,
+    ) -> impl Future<Output = Result<Option<Self::Output>, Error>> + Send + 'fut
+    where
+        C: 'fut,
+    {
+        ctx.parallel_join(zip(self, repeat(ctx.clone())).enumerate().map(
+            |(i, (bit, ctx))| async move {
+                generic_reveal(
+                    ctx.narrow(&TwoHundredFiftySixBitOpStep::Bit(i)),
```

Comment:
this is probably too specific for a generic reveal on a slice? Do you want to use `BitDecomposed` here or something specific to boolean shares maybe?

Reply:
I was thinking that within the scope of a single record, there couldn't be anything too big, but better just to implement it for `BitDecomposed`.